lizmap/modules/admin_api/lib/RepoCreator.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace LizmapApi;
+
+class RepoCreator
+{
+    /**
+     * Creates a new repository with the given properties.
+     *
+     * @param string $key                    The unique identifier for the repository. Must not yet exist.
+     * @param string $label                  the display name of the repository
+     * @param string $path                   the path to the repository directory
+     * @param bool   $allowUserDefinedThemes whether user-defined themes are allowed for the repository
+     * @param bool   $createDirectory        whether to create the repository directory
+     *
+     * @return bool returns true if the repository is successfully created
+     *
+     * @throws ApiException if the repository creation fails
+     */
+    public static function createRepository(
+        string $key,
+        string $label,
+        string $path,
+        bool $allowUserDefinedThemes,
+        bool $createDirectory
+    ): bool {
+        if (\lizmap::getRepository($key)) {
+            throw new ApiException("The repository '{$key}' already exists.", 409);
+        }
+
+        if (!$label) {
+            throw new ApiException('The repository label is not set.', 400);
+        }
+
+        if (!$path) {
+            throw new ApiException('The repository path is not set.', 400);
+        }
+
+        $data = array(
+            'label' => $label,
+            'path' => $path,
+            'allowUserDefinedThemes' => $allowUserDefinedThemes,
+        );
+
+        $rootRepositories = \lizmap::getServices()->getRootRepositories();
+
+        if ($rootRepositories == '' and $createDirectory) {
+            throw new ApiException('The root repository is not set, and you want to create a directory.', 403);
+        }
+
+        try {
+            $path = self::pathValidator($path, $rootRepositories);
+        } catch (ApiException $e) {
```

Comment:
I'm not sure, but you don't need to catch and resend as you are treating the same kind of exception `ApiException`. Up to you.